community/kernel/src/main/java/org/neo4j/helpers/Strings.java

```diff
@@ -74,4 +76,52 @@ public static String defaultIfBlank( String str, String defaultStr )
     {
         return isBlank( str ) ? defaultStr : str;
     }
+
+    public static String prettyPrint( Object o )
+    {
+        if ( o == null )
+        {
+            return "null";
+        }
+        else if ( o instanceof byte[] )
+        {
+            return Arrays.toString( (byte[]) o );
+        }
+        else if ( o instanceof short[] )
+        {
+            return Arrays.toString( (short[]) o );
+        }
+        else if ( o instanceof int[] )
+        {
+            return Arrays.toString( (int[]) o );
+        }
+        else if ( o instanceof long[] )
+        {
+            return Arrays.toString( (long[]) o );
+        }
+        else if ( o instanceof float[] )
+        {
+            return Arrays.toString( (float[]) o );
+        }
+        else if ( o instanceof double[] )
+        {
+            return Arrays.toString( (double[]) o );
+        }
+        else if ( o instanceof char[] )
+        {
+            return Arrays.toString( (char[]) o );
+        }
+        else if ( o instanceof boolean[] )
+        {
+            return Arrays.toString( (boolean[]) o );
+        }
+        else if ( o.getClass().isArray() )
+        {
+            return Arrays.deepToString( (Object[]) o );
+        }
```

Comment:
Perhaps do the `isArray` as an outer guard of all the other array checks, so we do fewer checks in the common case?